File: DSRFQ.Web/Modules/ServerTypes/Material/MaterialsColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { MaterialsRow } from './MaterialsRow';

export interface MaterialsColumns {
    Id: Column<MaterialsRow>;
    Code: Column<MaterialsRow>;
    Name: Column<MaterialsRow>;
    Description: Column<MaterialsRow>;
    Density: Column<MaterialsRow>;
    DimensionUnitId: Column<MaterialsRow>;
    CompanyId: Column<MaterialsRow>;
    InsertDate: Column<MaterialsRow>;
    InsertUserId: Column<MaterialsRow>;
    UpdateDate: Column<MaterialsRow>;
    UpdateUserId: Column<MaterialsRow>;
    DeleteDate: Column<MaterialsRow>;
    DeleteUserId: Column<MaterialsRow>;
    IsActive: Column<MaterialsRow>;
}

export class MaterialsColumns extends ColumnsBase<MaterialsRow> {
    static readonly columnsKey = 'Material.Materials';
    static readonly Fields = fieldsProxy<MaterialsColumns>();
}

File: DSRFQ.Web/Modules/Material/Materials/MaterialsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Material.Columns;

[ColumnsScript("Material.Materials")]
[BasedOnRow(typeof(MaterialsRow), CheckNames = true)]
public class MaterialsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Density { get; set; }
    public int DimensionUnitId { get; set; }
    public string CompanyName { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]

    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]

    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    [DateTimeFormatter(DisplayFormat = "yyyy-MM-dd HH:mm"), Width(120)]

    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}